src/dcqc/tests/base_test.py

```diff
@@ -195,12 +196,27 @@ def _interpret_process_outputs(self, outputs: dict[str, Path]) -> TestStatus:
         """Interpret the process output files to yield a test status."""
         exit_code = outputs["exit_code"].read_text()
         exit_code = exit_code.strip()
-        if exit_code == "0":
+        if exit_code == self.pass_code:
             status = TestStatus.PASS
         else:
             status = TestStatus.FAIL
         return status
 
+    # TODO: make changes to this package or the nextflow
+    # workflow so that file mounting is handled cleaner
+    @staticmethod
+    def _short_string_path(path: Path, substring: str) -> str:
+        # chech if substring is in path
+        if substring not in path.as_posix():
+            raise ValueError(f"{substring} not in {path}")
+        # get index where staged folder is
+        index = next(i for i, item in enumerate(path.parts) if substring in item)
```

Comment:
In the unlikely scenario there is something like:

`substring-A/substring-B/substring-C/your_file.txt`

That takes the first occurrence right? I think I'm ok with this. If the letters dcqc-staged ever gets arranged randomly in that order by nextflow, I'll just cry a little bit.

Reply:
Yes, this finds the first occurrence and then chops off anything that came before the substring. I think that is fine though because the only purpose of this string manipulation is to get us past the Nextflow-generated piece of the path. 

But yes, I suppose it is possible for Nextflow to by pure chance create a path with the substring in it. The way we are using it currently we are saved from this because we have a special character "-" in our substring which I don't think Nextflow would use in its generated paths.